Add constrain methods to UILayoutGuide

mc-vladyslav-kupriienko opened the pull request:
## Context

This PR adds the `constrain` methods to `UILayoutGuide`. 

This might be useful for scroll views for example, where we sometimes have to operate on `contentLayoutGuide` and `frameLayoutGuide`.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation including README.md
- [x] My changes generate no new warnings
- [x] My changes generate no breaking changes

mc-vladyslav-kupriienko committed:
Add constrain methods to UILayoutGuide

kuldar-daniel commented on Sources/MoonPresentation/LayoutConstraints/UIView+Constrainable.swift:
```diff
@@ -105,7 +105,7 @@ public extension UIView {
 }
 
-private extension NSLayoutConstraint.Edge {
+internal extension NSLayoutConstraint.Edge {
```
I think it would now make sense to move these extensions into separate files.

kuldar-daniel commented on Sources/MoonPresentation/LayoutConstraints/UIView+Constrainable.swift:
```diff
@@ -121,7 +121,7 @@ private extension NSLayoutConstraint.Edge {
 }
 
-private extension NSLayoutConstraint.Dimension {
+internal extension NSLayoutConstraint.Dimension {
```
Same as above.

kuldar-daniel commented on Sources/MoonPresentation/LayoutConstraints/UIView+Constrainable.swift:
```diff
@@ -133,7 +133,7 @@ private extension NSLayoutConstraint.Dimension {
 }
 
-private extension NSLayoutConstraint.Axis {
+internal extension NSLayoutConstraint.Axis {
```
Same as above.

kuldar-daniel commented on Sources/MoonPresentation/LayoutConstraints/UIView+Constrainable.swift:
```diff
@@ -147,7 +147,7 @@ private extension NSLayoutConstraint.Axis {
 }
 
-private extension CGFloat {
+internal extension CGFloat {
```
Same as above.

kuldar-daniel commented on Sources/MoonPresentation/LayoutConstraints/UIView+Constrainable.swift:
```diff
@@ -159,7 +159,7 @@ private extension CGFloat {
 }
 
-private extension UIEdgeInsets {
+internal extension UIEdgeInsets {
```
Same as above.

mc-vladyslav-kupriienko committed:
Move NSLayoutConstraint extensions

mc-vladyslav-kupriienko commented:
I have moved the `NSLayoutConstraint.*` and other related extensions to a separate file. I have also managed to unify the `UIView` and `UILayoutGuide` duplicate code by introducing a new function to the `Constrainable` protocol and moving the implementation to that protocol's extension.

I also ran SwiftFormat and SwiftLint on the package and fixed the warnings. I feel like it would make sense for our toolkit to follow our own SF/SL rules. I am torn on whether I should put it as a separate PR or just add to this one. Would love to hear your opinions!
